remote-api/src/main/java/org/alfresco/rest/api/impl/GroupsImpl.java

```diff
@@ -584,6 +613,13 @@ private Group getGroup(AuthorityInfo authorityInfo, List<String> includeParam, S
         group.setDisplayName(authorityDisplayName);
 
         group.setIsRoot(isRootAuthority(rootAuthorities, authorityInfo.getAuthorityName()));
+        group.setHasSubgroups(!authorityService.getContainedAuthorities(AuthorityType.GROUP, authorityInfo.getAuthorityName(), true).isEmpty());

```

Comment:
I'm pretty sure that code added in lines 616-621 is causing the tests to fail. 
I've seen that `authorityService.getContainedAuthorities` as well as `nodeService.getProperty` can throw an exception which is not handled.
Secondly, calling `isEmpty()` on `Set` can cause NPE (line 616) - it's much safer to use `CollectionUtils.isNotEmpty` there.
I'll try to come with some more findings soon as I am almost ready with hints on how to bring this closer to a working solution.

Reply:
Really good catch thanks! Added required changes